* grobian@gentoo.org: in particular -G breaks with gcc/gnu ld

--- config.mk.in
+++ config.mk.in
@@ -201,10 +201,8 @@
   SHLIB_SUFFIX = so
   shlibfn = $(1:%=%.$(SHLIB_SUFFIX).$(MAJ).$(MIN))
   shliblefn = $(1:%=%.$(SHLIB_SUFFIX))
-  # Solaris compiler (Sun C 5.5) can't take multiple ld options as
-  # -Wl,-a,-b .  Ld sees -a,-b in that case.
-  LDFLAGS_SHLIB = -Wl,-Bdynamic -Wl,-G -Wl,-h -Wl,$(SONAME)
-  CFLAGS_SHLIB = -Kpic
+  LDFLAGS_SHLIB = -shared -Wl,-soname,$(SONAME)
+  CFLAGS_SHLIB = -fPIC
 endif
 
 ifeq ($(HOST_OS),aix)
